### ISSUE TITLE
User focus can escape Dialog when navigating with tab

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#1554]
### Provide a general summary of the issue here

When using keyboard or screen reader and navigating with tab, the modal dialog is not a closed environment. When using keyboard or screen reader, the focus in an element with the role “dialog” should be contained in the element. Though a bit counterintuitive, this also includes the browser elements. 

### 🤔 Expected Behavior?

When using keyboard or screen reader and navigating with tab, the focus stays inside the modal, and doesn’t “leak” outside of it to the browser elements or the web page below the modal.

### 😯 Current Behavior

The user can access the browser elements and the page below when a modal is open and using tab.

### 💁 Possible Solution

When in a modal the focus should be locked using the following method:
•	Find the first and last focusable element programmatically
•	When user has focus in the last element of the modal, if the user presses the “Tab”-key, move focus to the first focusable element
•	When user has focus in the first element of the modal, if the user pressed “Shift + Tab” key combination, move focus to the last focusable element

### 🖥️ Steps to Reproduce

1. Open a page with the modal element (e.g. [Dialog Basic](Open https://storybook.sanomalearning.design/?path=/story/overlay-dialog--basic)
2. Use the tab key (or shift+tab to reverse) to navigate the element (optionally start a screen reader)
3. When in the first or last focusable element of the modal, keep pressing tab (or shift-tab) to escape the modal

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 10

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

### Tasks
- [ ] https://github.com/sl-design-system/components/issues/1555
- [ ] https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#accessibility
- [ ] https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/
- [ ] update statement in issue
- [ ] add deviations in wcag messege dialog page and explain why we don't follow the 'wcag rules'
- [x] Discuss with Valterri in upcoming meeting
- [x] implement focus trap within dialog

**Comment by anqaka** [comment]
Is it worth to use popover attribute for dialog? https://hidde.blog/popover-accessibility/

**Comment by jpzwarte** [comment]
I'm reopening this to reevaluate the decision to use a 3rd party library for focus trapping in `<dialog>`. The reason is that WHATWG, together with W3C's Accessible Platform Architectures (APA) WG concluded that it isn't necessary and that the current native dialog element behavior is OK.

See https://github.com/whatwg/html/issues/8339#issuecomment-1822591131

**Comment by jpzwarte** [comment]
To add to this, there's an issue to improve the WCAG docs on this point: https://github.com/w3c/wcag/issues/4185

**Comment by jpzwarte** [comment]
As decided during refinement of the reopened issue: we will remove the focus trapping behavior of dialog and allow the users to focus browser UI while a modal `<dialog>` is open.
